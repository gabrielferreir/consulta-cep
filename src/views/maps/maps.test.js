import React from 'react';
import ConnectMaps, {Maps, Icon} from './maps';
import thunk from 'redux-thunk';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {configure} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import configureStore from 'redux-mock-store';

const mockStore = configureStore([thunk]);
const initialState = {};

let store;

beforeEach(() => {
    configure({adapter: new Adapter()});
    store = mockStore(initialState);
});

it('Deve renderizar o mapa storie', () => {
    const wrapper = shallow(<ConnectMaps store={store}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
});

it('Deve renderizar a mensagem de loading', () => {
    const wrapper = shallow(<Maps opened={false} isLoading={true}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
});

it('Deve renderizar com o mapa', () => {
    const address = {
        "cep": "14402-336",
        "logradouro": "Rua Francisco Miglioranza",
        "complemento": "",
        "bairro": "Jardim Palestina",
        "localidade": "Franca",
        "uf": "SP",
        "unidade": "",
        "ibge": "3516200",
        "gia": "3104",
        "lat": -20.5353875,
        "lng": -47.3588774
    };
    const wrapper = shallow(<Maps opened={true} address={address}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
});

it('Deve renderizar o icone', () => {
    const icon = shallow(<Icon/>);
    expect(toJson(icon)).toMatchSnapshot();
});

it('Deve renderizar vazio', () => {
    const wrapper = shallow(<Maps opened={true} isLoading={true}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
});