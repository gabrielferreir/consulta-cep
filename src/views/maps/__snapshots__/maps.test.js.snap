// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar a mensagem de loading 1`] = `
<div
  style={
    Object {
      "padding": "16px",
    }
  }
>
  <p>
    Carregando...
  </p>
</div>
`;

exports[`Deve renderizar com o mapa 1`] = `
<section
  className="wrapper-map"
>
  <div
    className="wrapper-map__clear"
  >
    <i
      className="material-icons"
    >
      clear
    </i>
  </div>
  <h1
    className="wrapper-map__title"
  >
    Rua Francisco Miglioranza
  </h1>
  <p
    className="wrapper-map__info-address"
  >
    Jardim Palestina
  </p>
  <p
    className="wrapper-map__info-address"
  >
    Franca
     - 
    SP
  </p>
  <p
    className="wrapper-map__info-address"
  >
    14402-336
  </p>
  <div
    className="wrapper-map__map"
  >
    <GoogleMap
      bootstrapURLKeys={
        Object {
          "key": "teste123",
        }
      }
      debounced={true}
      defaultCenter={
        Object {
          "lat": -20.5353875,
          "lng": -47.3588774,
        }
      }
      defaultZoom={18}
      distanceToMouse={[Function]}
      googleMapLoader={[Function]}
      heatmap={Object {}}
      heatmapLibrary={false}
      hoverDistance={30}
      layerTypes={Array []}
      options={[Function]}
      shouldUnregisterMapOnUnmount={true}
      style={
        Object {
          "height": "100%",
          "margin": 0,
          "padding": 0,
          "position": "relative",
          "width": "100%",
        }
      }
      yesIWantToUseGoogleMapApiInternals={false}
    >
      <Icon
        lat={-20.5353875}
        lng={-47.3588774}
      />
    </GoogleMap>
  </div>
</section>
`;

exports[`Deve renderizar com o mapa 2`] = `<div />`;

exports[`Deve renderizar o icone 1`] = `
<img
  alt="location"
  src="icon-location.svg"
/>
`;

exports[`Deve renderizar o mapa 1`] = `
<ContextProvider
  value={null}
>
  <Maps
    clear={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;

exports[`Deve renderizar o mapa storie 1`] = `
<ContextProvider
  value={null}
>
  <Maps
    clear={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;

exports[`Deve renderizar vazio 1`] = `<div />`;
